scripts/qemu-check.sh

```diff
@@ -7,110 +7,133 @@ READY_MSG="6c 00 00 00 00 00 00 70"
 
 rm -f dump-*.txt
 
-if [ "$#" -eq 0 ]
-then
-	PLATFORMS=${SUPPORTED_PLATFORMS[@]}
+print_usage()
+{
+	cat <<EOF
+usage: qemu-check.sh platform(s)
+	Supported platforms are ${SUPPORTED_PLATFORMS[*]}
+EOF
+}
+
+while getopts "" OPTION; do
+	case "$OPTION" in
+		*) print_usage; exit 1 ;;
+	esac
+done
+shift $((OPTIND-1))
+
+PLATFORMS=()
+if [ "$#" -eq 0 ]; then
+	PLATFORMS=("${SUPPORTED_PLATFORMS[@]}")
 else
-	for args in $@
-	do
-		for i in ${SUPPORTED_PLATFORMS[@]}
-		do
-			if [ $i == $args ]
-			then
-				PLATFORMS+=$i" "
+	for arg in "$@"; do
+		platform=none
+		for i in "${SUPPORTED_PLATFORMS[@]}"; do
+			if [ x"$i" = x"$arg" ]; then
+				PLATFORMS=("${PLATFORMS[@]}" "$i")
+				platform=$i
+				shift
+				break
 			fi
 		done
+		if [ "$platform" = "none" ]; then
+			echo "Error: Unknown platform specified: $arg"
+			echo "Supported platforms: ${SUPPORTED_PLATFORMS[*]}"
+			exit 1
```

Comment:
When there is more than one `echo ERRROR; exit 1` you should (re-)use a `die()` function like this one:
```shell
die()
{
    >&2 printf '%s: ERROR: ' "$0"
    >&2 printf "$@"
    exit 1
}
```
Then you can shorten the input validation checks into one-liners like this:
```shell
[ ${#PLATFORMS[@]} -gt 0 ] || die  ''No platforms specified. Supported are: %s\n'\
		"${SUPPORTED_PLATFORMS[*]}"
```

Reply:
After i remove the exit for no platforms which is unreachable, this would be the only one